packages/admin-ui-docs/docs/guidelines/components/switch.mdx

```diff
@@ -0,0 +1,284 @@
+    [
+      'Usage',
+      'State',
+      'checked and onChange',
+      'useSwitchState',
+      'Props',
+    ].includes(item.value)
+  ),
+  designReference: toc.filter((item) => ['Do’s'].includes(item.value)),
+}
+
+<Page toc={tabToc} hasTabs>
+  <Panel id="overview">
+
+Switch is a component for customizing configurations. It represents a physical switch that allows users to turn things on or off, providing control over actions and preferences. Tapping a toggle switch is a two-step action: selection and execution. It should reflect an action that triggers a state, never both at once.
+
+## Example
+
+```jsx live
+function Example() {
+  const [checked, setChecked] = React.useState(false)
+
+  return (
+    <Switch
+      label="Label"
+      ariaLabel="label"
+      state={{
+        checked,
+        onChange: () => setChecked(!checked),
+      }}
+    />
+  )
+}
+```
+
+## Variations
+
+### Standalone
+
+When using a standalone `Switch` you should provide an `aria-label` property value. As a `form` component, it should have a label specified to guarantee accessibility.
+
+```jsx live
+function Example() {
+  const [checked, setChecked] = React.useState(false)
+
+  return (
+    <Switch
+      label="Label"
+      aria-label="label"
+      checked={checked}
+      onChange={() => setChecked(!checked)}
+    />
+  )
+}
+```
+
+### With help text
+
+```jsx live
+function Example() {
+  const [checked, setChecked] = React.useState(false)
+
+  return (
+    <Switch
+      label="Label"
+      aria-label="label"
+      checked={checked}
+      helpText="Help text"
+      onChange={() => setChecked(!checked)}
+    />
+  )
+}
+```
+
+### With error text
+
+```jsx live
+function Example() {
+  const [checked, setChecked] = React.useState(false)
+
+  return (
+    <Switch
+      label="Label"
+      aria-label="label"
+      checked={checked}
+      error
+      errorText="Error text"
+      onChange={() => setChecked(!checked)}
+    />
+  )
+}
+```
+
+### With help and error texts
+
+```jsx live
+function Example() {
+  const [checked, setChecked] = React.useState(false)
+
+  return (
+    <Switch
+      label="Label"
+      aria-label="label"
+      checked={checked}
+      error
+      errorText="Error text"
+      helpText="Help text"
+      onChange={() => setChecked(!checked)}
+    />
+  )
+}
+```
+
+### Disabled
+
+```jsx live
+function Example() {
+  return (
+    <>
+      <Switch
+        label="Label"
+        aria-label="label"
+        error
+        errorText="Error text"
+        helpText="Help text"
+        onChange={() => setChecked(!checked)}
+        disabled
+      />
+      <Switch
+        label="Label"
+        aria-label="label"
+        error
+        checked
+        errorText="Error text"
+        helpText="Help text"
+        onChange={() => setChecked(!checked)}
+        disabled
+      />
+    </>
+  )
+}
+```
+
+### Checked States
+
+There are two checked states: `not checked`, and `checked`.
```

Comment:
Nitpick, start with "checked" and then mention "not checked". "Checked" makes more sense as a first option.